### PR TITLE
IGNITE-13829: Added log rotation to ducktape-tests

### PR DESCRIPTION
**Pull request by Sega76**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [x] There is a single JIRA ticket related to the pull request. 
- [x] The web-link to the pull request is attached to the JIRA ticket.
- [x] The JIRA ticket has the _Patch Available_ state.
- [x] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/ducktests/tests/ignitetest/services/ignite.py [hunk 1]
```diff
@@ -86,4 +86,4 @@ def get_event_time(service, log_node, log_pattern, from_the_beginning=True, time
         "grep '%s' %s" % (log_pattern, IgniteAwareService.STDOUT_STDERR_CAPTURE))
 
     return datetime.strptime(re.match("^\\[[^\\[]+\\]", stdout.read().decode("utf-8")).group(),
-                             "[%Y-%m-%d %H:%M:%S,%f]")
+                             "[%Y-%m-%dT%H:%M:%S,%f]")
```
Comment (anton-vinogradov) [inline modules/ducktests/tests/ignitetest/services/ignite.py:89]: why changed?

File: modules/ducktests/tests/ignitetest/services/utils/ignite_spec.py [hunk 2]
```diff
@@ -102,12 +102,13 @@ class IgniteNodeSpec(IgniteSpec, IgnitePersistenceAware):
...
     def command(self):
-        cmd = "%s %s %s %s 2>&1 | tee -a %s &" % \
+        cmd = "%s %s %s %s 2>&1 | tee %s >> %s &" % \
               (self._envs(),
                self.path.script("ignite.sh"),
                self._jvm_opts(),
                self.CONFIG_FILE,
-               self.STDOUT_STDERR_CAPTURE)
+               self.STDOUT_STDERR_CAPTURE,
+               self.STDOUT_STDERR_ALL)
```
[the remark this answers is not in the document]
Comment (Sega76) [reply modules/ducktests/tests/ignitetest/services/utils/ignite_spec.py:111]: console.log is used as a monitor, when the service starts, we use it to determine that it has started.
It appears earlier than ignite.log. on restart, if we look at the old log, we will see all the events there, and we will not wait for them. it is not right

console.log contains stdout the last start
console_all.log contains stdout all starts

File: modules/ducktests/tests/ignitetest/services/ignite.py (hunk 1 above)
Comment (Sega76) [reply modules/ducktests/tests/ignitetest/services/ignite.py:89]: format log4j2
example output:
[2020-12-09T19:10:58,878][INFO ][main][IgniteAwareApplicationService] Starting Application... [params=/mnt/service/ignite-config.xml]

File: modules/ducktests/tests/ignitetest/services/utils/ignite_spec.py [hunk 3]
```diff
@@ -186,8 +188,8 @@ def __init__(self, context, modules, servicejava_class_name, java_class_name, pa
 
         self.jvm_opts.extend([
             "-DIGNITE_SUCCESS_FILE=" + self.PERSISTENT_ROOT + "/success_file",
-            "-Dlog4j.configuration=file:" + self.LOG4J_CONFIG_FILE,
-            "-Dlog4j.configDebug=true",
+            "-Dlog4j2.debug",
+            "-Dlog4j2.configurationFile=" + self.PERSISTENT_ROOT + "/ignite-log4j2.xml",
```
[the remark this answers is not in the document]
Comment (Sega76) [reply modules/ducktests/tests/ignitetest/services/utils/ignite_spec.py]: moved to IgniteSpec

File: modules/ducktests/tests/ignitetest/services/utils/ignite_persistence.py [hunk 4]
```diff
@@ -43,8 +45,7 @@ def init_persistent(self, node):
         Init persistent directory.
         :param node: Service node.
         """
-        node.account.mkdirs(self.PERSISTENT_ROOT)
-        node.account.mkdirs(self.TEMP_DIR)
+        node.account.mkdirs(f'{self.PERSISTENT_ROOT} {self.TEMP_DIR} {self.LOGS_DIR}')
```
[the remark this answers is not in the document]
Comment (Sega76) [reply modules/ducktests/tests/ignitetest/services/utils/ignite_persistence.py]: fixed

File: modules/ducktests/tests/ignitetest/services/utils/ignite_spec.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (anton-vinogradov) [reply modules/ducktests/tests/ignitetest/services/utils/ignite_spec.py:117]: just set offset not to 0 but to the position related to the restart when from_the_beginning is true

File: modules/ducktests/tests/ignitetest/services/utils/ignite_spec.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (anton-vinogradov) [reply modules/ducktests/tests/ignitetest/services/utils/ignite_spec.py]: please make sure this does not break tests at SE plugin

File: modules/ducktests/tests/ignitetest/services/ignite.py (hunk 1 above)
Comment (anton-vinogradov) [reply modules/ducktests/tests/ignitetest/services/ignite.py:89]: no, my question is "why the format changed"

File: modules/ducktests/tests/ignitetest/services/utils/ignite_spec.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Sega76) [reply modules/ducktests/tests/ignitetest/services/utils/ignite_spec.py:117]: In a chat discussion, we came to this decision.
the monitor should be to one start.

[the remark this answers is not in the document]
Comment (timoninmaxim) [reply modules/ducktests/tests/ignitetest/services/utils/ignite_spec.py:117]: > just set offset not to 0 but to the position related to the restart when from_the_beginning is true

Hi! I suggest to use separate file for last run for monitor purposes. Setting offset will require getting the offset from somewhere, storing that. Also there could be issues with unplanned restarts of a service. All this is an additional code, checks and we can just use separate file.

File: modules/ducktests/tests/ignitetest/services/utils/ignite_spec.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (timoninmaxim) [reply modules/ducktests/tests/ignitetest/services/utils/ignite_spec.py]: Duplication is OK for spec for explicit view of start command line. Also in future spec may be configured with files and there will be a question how to merge specs with common part etc. So let's duplicate it for a while.

[the remark this answers is not in the document]
Comment (Sega76) [reply modules/ducktests/tests/ignitetest/services/utils/ignite_spec.py:192]: Discussed with Maxim Timonin, returned this changes

File: modules/ducktests/tests/ignitetest/services/ignite.py (hunk 1 above)
Comment (Sega76) [reply modules/ducktests/tests/ignitetest/services/ignite.py:89]: format ISO8601:
https://en.wikipedia.org/wiki/ISO_8601

before it was not applied

Comment (timoninmaxim) [reply modules/ducktests/tests/ignitetest/services/ignite.py:89]: From your link:

>  single point in time can be represented by concatenating a complete date expression, the letter "T" as a delimiter, and a valid time expression. For example, "2007-04-05T14:30". In ISO 8601:2004 it was permitted to omit the "T" character by mutual agreement as in "200704051430", but this provision was removed in ISO 8601-1:2019. **Separating date and time parts with other characters such as space is not allowed in ISO 8601, but allowed in its profile RFC 3339**.

As I know RFC 3339 is wide adopted too. Actually do we need care about it?

**Pull request closed without merging**